keychain_mac.cpp

```diff
@@ -89,10 +89,24 @@ void updateError(keychain::Error &err, OSStatus status) {
     }
 
     err.message = errorStatusToString(status);
-    err.code = status; // TODO check conversion
-    err.error = status == errSecItemNotFound
-                    ? keychain::KeychainError::NotFound
-                    : keychain::KeychainError::GenericError;
+    err.code = status;
+
+    switch (status) {
+    case errSecItemNotFound:
+        err.error = keychain::KeychainError::NotFound;
+        break;
+
+    // potential errors in case the user needs to unlock the keychain first
+    case errSecUserCanceled:        // user pressed the Cancel button
+    case errSecAuthFailed:          // too many failed password attempts
+    case errSecInteractionRequired: // user interaction required but not allowed
+        err.error = keychain::KeychainError::AccessDenied;
+        break;
+
+    case errSecDataTooLarge:  // TODO: == PasswordTooLong?
```

Comment:
I'll remove this one again. Even with _very_ large service name, account name and password no error was reported. It just took quite long to run the tests. So I don't know how to trigger this.